kod/util/rntmaint.kod

```diff
@@ -74,6 +100,150 @@ messages:
       return;
    }
 
+   ParseInnkeeperCommands(who=$, what=$, innRid=$, string=$,
+      initCost=ROOM_INIT_RENT_COST_DEFAULT, perDayCost=ROOM_PER_DAY_COST_DEFAULT, keyCopyCost=ROOM_KEY_COPY_COST_DEFAULT,
+      rscRoomRented=$, rscRoomCost=$)
+   {
+      local rSayRsc;
+
+      % If it's not a player talking, we don't want to do anything else here.  Abort.
+      if NOT IsClass(what, &User)
+      {
+         return;
+      }
+
+      % These values are the bare minimum that should be passed in from
+      % a listening innkeeper.  If they're not all present, abort and
+      % leave a console message.
+      if who = $ OR innRid = $ OR string = $
+      {
+         Debug("Warning: ParseInnkeeperCommands is missing required parameters: [#who, #innRid, #string]");
+         Debug("#who:", who, " - innkeeper object");
+         Debug("#innRid:", innRid, " - inn's room id");
+         Debug("#string:", string, " - string spoken by player");
+         return;
+      }
+
+      % Any of these not specified can just use generic defaults
+      if rscRoomRented = $ { rscRoomRented = RoomMaintenance_innkeeper_say_room_rented; }
+      if rscRoomCost = $ { rscRoomCost = RoomMaintenance_innkeeper_say_cost; }
+
+      % Command:  Rent Room
+      if (StringContain(string, RoomMaintenance_command_rent) AND StringContain(string, RoomMaintenance_command_room))
+         OR (StringContain(string, "rent") AND StringContain(string, "room"))
+      {
+         rSayRsc = Send(self, @RentRoom, #who=what, #iLocation=innRid, #iCost=initCost);
+
+         % rSayRsc is $ on success.  Use the innkeeper's saying.
+         if rSayRsc = $
+         {
+            rSayRsc = rscRoomRented;
+         }
+
+         Post(Send(who, @GetOwner), @SomeoneSaid, #type=SAY_RESOURCE, #what=who, #string=rSayRsc);
+
+         return;
+      }          
+
+      % Command: Copy Key
+      if StringContain(string, RoomMaintenance_command_copy_key)
+         OR StringContain(string,"copy key")
+      {
+         rSayRsc = send(self, @CopyKey, #who=what, #iLocation=innRid, #iCost=keyCopyCost);
+
+         % tell the player.
+         Post(Send(who, @GetOwner), @SomeoneSaid, #type=SAY_RESOURCE, #what=who, #string=rSayRsc);
+
+         return;
+      }
+
+      % Command: Room Cost
+      if (StringContain(string, RoomMaintenance_command_room) AND StringContain(string, RoomMaintenance_command_cost))
+         OR (StringContain(string, "room") AND StringContain(string, "cost"))
+      {
+         Post(Send(who, @GetOwner), @SomeoneSaid, #type=SAY_RESOURCE, #what=who,
+            #string=rscRoomCost, 
+            #parm1=initCost, #parm2=perDayCost, #parm3=keyCopyCost);
+
+         return;
+      }    
+
+      return;
+   }
+
+   % Checks items offered (ReqOffer) to an innkeeper to see if they're related to rooms
```

Comment:
Period at the end of the sentence.

Make this into a doc string rather than a comment?  That way it's readable from admin mode.